File: intro/go/maxvalue.go

package main

import (
	"fmt"
	"math"
)

func maxValue(numbers []float64) float64 {
	max := math.Inf(-1)
	for _, num := range numbers {
		if num > max {
			max = num
		}
	}
	return max
}

func main() {
	fmt.Println(maxValue([]float64{1.2, 5.6, 3.4})) // 5.6
	fmt.Println(maxValue([]float64{}))              // -Inf
}

File: intro/php/max_value.php
<?php

declare(strict_types=1);


function maxValue(array $numbers): float {
    $max = -INF;
    foreach ($numbers as $num) {
        if ($num > $max) {
            $max = $num;
        }
    }
    return $max;
}


if (basename(__FILE__) === basename($_SERVER['SCRIPT_FILENAME'] ?? '')) {
    echo maxValue([1.2, 5.6, 3.4]) . PHP_EOL; // 5.6
    echo maxValue([]) . PHP_EOL;              // -INF
}